drivers/serial: stm32: Modify Kconfig instance flags

erwango opened the pull request:
STM32 uart Kconfig instance flags were not following
same naming scheme than other drivers (i2c, spi, ..)

Update driver to use UART_X instead of UART_STM32_PORT_X

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
Shippable timed out.
I tested beforehand with sanitycheck with all boards on hello_world. 
So I suspect timeout is due to impact of change..
Let my try once more

erwango commented:
recheck

erwango commented:
recheck

nashif commented on boards/arm/96b_neonkey/Kconfig.defconfig:
```diff
@@ -12,8 +12,8 @@ config BOARD
-config UART_STM32_PORT_1
-	default y
+config UART_1
+	def_bool y
```
default y, see 46386522142e546c6e028db0f560c6db25a02d06

erwango replied:
Txs, rebase issue I guess. Fixed

erwango commented:
@nashif, comments addressed

erwango committed:
drivers/serial: stm32: Modify Kconfig instance flags

STM32 uart Kconfig instance flags were not following
same naming scheme than other drivers (i2c, spi, ..)

Update driver to use UART_X instead of UART_STM32_PORT_X

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
